### PR TITLE
feat(code-gen-types): add createResourceDbContextFile plugin event

### PR DESCRIPTION
**Pull request by morhag90**
Close: #8577 
## PR Details

Add createResourceDbContextFile plugin event. 

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: libs/util/code-gen-types/src/dotnet-plugin-events-params.types.ts [hunk 1]
```diff
@@ -63,6 +63,12 @@ export interface CreateEntityModelParams extends EventParams {
   apisDir: string;
 }
 
+export interface createResourceDbContextFileParams extends EventParams {
+  entities: Entity[];
+  resourceName: string;
+  apisDir: string;
```
Comment (overbit) [inline libs/util/code-gen-types/src/dotnet-plugin-events-params.types.ts:69]: Is `apisDir` needed? Or should it be just context folder?

Comment (morhag90) [reply libs/util/code-gen-types/src/dotnet-plugin-events-params.types.ts:69]: @overbit  We are passing it to the internal method, so I think we need to pass it to the plugin event too:

image

Comment (overbit) [reply libs/util/code-gen-types/src/dotnet-plugin-events-params.types.ts:69]: Let's use the same name then. Apisdir is a bit cryptic

Comment (morhag90) [reply libs/util/code-gen-types/src/dotnet-plugin-events-params.types.ts]: Done

**Review by overbit — APPROVED** [review]

**Pull request merged**